perceptia-one/perceptia/src/components/QuestionView.js

```diff
@@ -0,0 +1,158 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import Button from '@material-ui/core/Button';
+import Card from '@material-ui/core/Card';
+import CardActions from '@material-ui/core/CardActions';
+import CardContent from '@material-ui/core/CardContent';
+import CardMedia from '@material-ui/core/CardMedia';
+import CssBaseline from '@material-ui/core/CssBaseline';
+import Grid from '@material-ui/core/Grid';
+import Toolbar from '@material-ui/core/Toolbar';
+import Typography from '@material-ui/core/Typography';
+import { withStyles } from '@material-ui/core/styles';
+import { BrowserRouter as Router, Route, Link } from "react-router-dom";
+import QuizInfo from './QuizInfo';
+import Question from './Question';
+import constants from './constants';
+import Results from './Results'
+import './quiz.css';
+
+
+class QuestionView extends React.Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            questions: [],
+            question: null,
+            questionId: null,
+            question_type: null,
+            answerOptions: [],
+            answers: {},
+            counter: 0,
+            completed: false,
+
+        }
+    }
+
+    componentDidMount() {
+
+        const { selectedQuiz } = this.props.location.state
+
+        fetch(constants.localhost.url + 'questions/' + selectedQuiz)
```

Comment:
It does not look like there is any error handling for 4xx and 5xx http status codes, this should be handled here.